src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -295,15 +297,23 @@ public void close() throws ConnectException {
           commitFile(encodedPartition);
         }
       } catch (IOException e) {
-        log.error("Error rotating {} when closing TopicPartitionWriter.", tempFiles.get(encodedPartition));
+        log.error("Error rotating {} when closing TopicPartitionWriter.",
+                  tempFiles.get(encodedPartition));
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(baos);
+        e.printStackTrace(ps);
+        ps.close();
+        log.error(baos.toString());
+        log.error(e.getMessage());
+        log.error(e.toString());
         exceptions.add(e);
       }
-    }
     writers.clear();
     try {
       wal.close();
     } catch (ConnectException e) {
       log.error("Error closing {}.", wal.getLogFile());
+      log.debug(e.getMessage());
```

Comment:
Same here, if we're logging the error, it's probably fine to log the exception that caused it too.